src/observers/InformationMessageObserver.ts

```diff
@@ -0,0 +1,35 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as ObservableEvent from "../omnisharp/loggingEvents";
+
+import { dotnetRestoreForProject } from '../features/commands';
+import { vscode } from '../vscodeAdapter';
+
+export class InformationMessageObserver {
+    constructor(private vscode: vscode) {
+    }
+
+    public post = (event: ObservableEvent.BaseEvent) => {
+        switch (event.constructor.name) {
+            case ObservableEvent.OmnisharpServerUnresolvedDependencies.name:
+            this.handleOmnisharpServerUnresolvedDependencies(<ObservableEvent.OmnisharpServerUnresolvedDependencies>event);
+            break;
+        }
+    }
+
+    private handleOmnisharpServerUnresolvedDependencies(event: ObservableEvent.OmnisharpServerUnresolvedDependencies) {
```

Comment:
FWIW, it'd probably be a good idea to specify return types in general, but especially when a random Promise is the result. It's really unclear from the signature what this is expected to return, which makes the code hard to understand from a cursory read.